pandas/core/strings/object_array.py

```diff
@@ -61,9 +61,12 @@ def _str_map(self, f, na_value=None, dtype: Optional[Dtype] = None):
             na_value = self._str_na_value
 
         if not len(self):
-            # error: Argument 1 to "ndarray" has incompatible type "int";
-            # expected "Sequence[int]"
-            return np.ndarray(0, dtype=dtype)  # type: ignore[arg-type]
+            # error: Argument "dtype" to "array" has incompatible type
+            # "Union[ExtensionDtype, str, dtype[Any], Type[object]]"; expected
+            # "Union[dtype[Any], None, type, _SupportsDType, str,
+            # Union[Tuple[Any, int], Tuple[Any, Union[int, Sequence[int]]], List[Any],
+            # _DTypeDict, Tuple[Any, Any]]]"
+            return np.array([], dtype=dtype)  # type: ignore[arg-type]
```

Comment:
I'm confused about why `mypy` reports this, if I try making a file `t.py` with
```python
import numpy as np
from pandas._typing import Dtype, Optional

def foo(dtype: Optional[Dtype] = None):
    if dtype is None:
        dtype = np.dtype("object")
    return (np.array([], dtype=type))
```
then `mypy` type checks it just fine. Do you know what I might be missing?

Reply:
no idea.  i think this edit is unrelated to most of the rest; using `np.ndarray` here instead of `np.array`  weirds me out

Reply:
The `Dtype` alias includes `ExtensionDtype` which cannot be passed on to numpy

```
>>> np.array([], dtype=pd.Int64Dtype())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Cannot interpret 'Int64Dtype()' as a data type
>>> 
```

> Do you know what I might be missing?

`return (np.array([], dtype=type))` -> `return (np.array([], dtype=dtype))`

Reply:
I'm embarrassed, thanks!

Reply:
so this looks like there maybe an actual bug here. As part of the ArrowStringArray work, I have been parameterising existing tests on object dtype arrays with StringArray and ArrowStringArray and this has maybe uncovered some latent bugs with StringArray.

I think OK to leave this ignore as a 'fix later' and out of scope here.